process/block/pendingMb/pendingMiniBlocks.go

```diff
@@ -29,10 +29,7 @@ func (p *pendingMiniBlocks) getAllCrossShardMiniBlocksHashes(metaBlock *block.Me
 	crossShardMiniBlocks := make(map[string]uint32)
 
 	for _, mbHeader := range metaBlock.MiniBlockHeaders {
-		if mbHeader.SenderShardID != core.MetachainShardId && mbHeader.ReceiverShardID == core.MetachainShardId {
-			continue
-		}
-		if mbHeader.SenderShardID == core.MetachainShardId && mbHeader.ReceiverShardID == core.AllShardId {
+		if !shouldConsiderCrossShardMiniBlock(mbHeader.SenderShardID, mbHeader.ReceiverShardID) {
```

Comment:
these conditions are not equal to the old variant. Is this intended?

Reply:
Actually these conditions should been applied in the same way in both iterations. The basic idea with method getAllCrossShardMiniBlocksHashes is that it extracts from one metablock all the cross shard miniblocks (dst != rcv no matter which is dst or rcv) which would appear 2 times in a while. First time when you see it you would add it as pending miniblock and the second time you would removed it from pending. So, there are some exception when a cross mini block would not appear 2 times. and these cases are: 1) system smart contract (miniblocks from shard to meta) and 2) PeerBlock miniblocks (from meta to AllShardID)

Reply:
ok